src/evaluators.js

```diff
@@ -155,6 +140,27 @@ export function createSafeEvaluatorFactory(unsafeRec, safeGlobal, transforms) {
     // of Function in any compartment of the same root realm.
     setPrototypeOf(safeEval, unsafeFunction.prototype);
 
+    // todo (shim limitation): scan endowments, throw error if endowment
+    // overlaps with the const optimization (which would otherwise
+    // incorrectly shadow endowments), or if endowments includes 'eval'. Also
+    // prohibit accessor properties (to be able to consistently explain
+    // things in terms of shimming the global lexical scope).
+    // writeable-vs-nonwritable == let-vs-const, but there's no
+    // global-lexical-scope equivalent of an accessor, outside what we can
+    // explain/spec
+    const scopeTarget = create(safeGlobal, {
+      ...getOwnPropertyDescriptors(endowments),
+      eval: {
```

Comment:
Did you mean to have this `eval` override what may have been contributed as an `eval` in the endowments? I think it's better to enable the endowments to override this setting, in which case `eval: safeEval` would be an overridable default.

I do see in the todo comments above that we should ban `eval` from the endowments, in which case this isn't an observable issue. However, I can no longer reconstruct why we should prohibit it. 

Did we not put `eval: safeEval` on the `safeGlobal`? Recently, I've been thinking that the `scopeTarget` is a better place anyway, for both the safe `eval` and the safe `Function` constructor. But that has not been the plan. Does the safe `Function` constructor appear on the `safeGlobal`? In any case, they should both appear in the same place.

Thus, if we do place `Function` on the `scopeTarget`, it should go either before or after the endowments, in the same position as `eval`.

Reply:
Today I experimented with putting safe `eval` and `Function` constructor on the `scopeTarget`.  Some tests are now failing because they expect to do `r.global.Function(...)` or `new r.global.Function(...)`, and cannot find the `Function` constructor there.

What should be the expected behaviour?  I can certainly rewrite the tests to use `r.evaluate('Function')(...)` and `new (r.evaluate('Function'))(...)`, but I need to make sure the plan for this whole `scopeTarget` versus `safeGlobal` thing is fully-baked.